drivers/flash/flash_stm32_ospi.c

```diff
@@ -37,6 +44,27 @@ LOG_MODULE_REGISTER(flash_stm32_ospi, CONFIG_FLASH_LOG_LEVEL);
 #define STM32_OSPI_SUBSECTOR_4K_ERASE_MAX_TIME  400U
 #define STM32_OSPI_WRITE_REG_MAX_TIME           40U
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_dma) || DT_HAS_COMPAT_STATUS_OKAY(st_stm32_dmamux)
```

Comment:
dma could be enabled for other purpose than ospi.
What about the following:
`#if STM32_OSPI_USE_DMA`